cmd/release-controller/upgrades.go

```diff
@@ -70,6 +70,7 @@ func (g *UpgradeGraph) SummarizeUpgradesTo(toNames ...string) []UpgradeHistory {
 				Success: h.Success,
 				Failure: h.Failure,
 				Total:   len(h.History),
+				History: h.History,
```

Comment:
not safe, the history is under the lock to prevent concurrent modifications with reads.  you'll have to do something under the lock to simplify it, or copy it.  both are going to be expensive